### PR TITLE
Fix/Add allow_empty flag to randomT containers

### PR DESCRIPTION
**Pull request by lhruby**
# Description
Some tests use empty containers to test for corner cases, while in other tests, an empty container is not sensible. We add a flag to specify if an empty container is acceptable.

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [ ] If this is a new component I have added examples.
- [ ] I updated the README and related documentation.

### CODE REVIEW

File: modules/random/include/hephaestus/random/random_container.h [hunk 1]
```diff
@@ -4,15 +4,58 @@
...
+#include <fmt/format.h>
 #include <magic_enum.hpp>
 
 #include "hephaestus/random/random_type.h"
+#include "hephaestus/utils/exception.h"
 
 namespace heph::random {
 
+namespace internal {
+[[nodiscard]] inline auto getSize(std::mt19937_64& mt, std::optional<size_t> size,
```
[the remark this answers is not in the document]
Comment (lhruby) [reply modules/random/include/hephaestus/random/random_container.h]: Implemented
